preview/Win7Msix/Win7MSIXInstaller/ValidateRequest.cpp

```diff
@@ -0,0 +1,88 @@
+#include <windows.h>
+
+#include <shlobj_core.h>
+#include <CommCtrl.h>
+
+#include "FilePaths.hpp"
+#include "ValidateRequest.hpp"
+#include "GeneralUtil.hpp"
+#include <TraceLoggingProvider.h>
+#include <VersionHelpers.h>
+
+const PCWSTR ValidateRequest::HandlerName = L"ValidateRequest";
+
+HRESULT ValidateRequest::ExecuteForAddRequest()
+{
+    RETURN_IF_FAILED(ParseDependencyInfoFromPackage());
+    RETURN_IF_FAILED(MatchTDFAndVersionToOS());
+    return S_OK;
+}
+
+HRESULT ValidateRequest::ParseDependencyInfoFromPackage()

```

Comment:
>ValidateRequest [](start = 8, length = 15)

i'd want this to be more specific. ValidateTargetDeviceFamily?

Reply:
Since this method is just parsing the manifest and fetching the fields and not validating the values yet, would ParseTargetDeviceFamilyFields be more accurate?

Reply:
yeah method name mentioning TargetDeviceFamily is a better name than DependencyInfo, but my original comment was referring to the name of the class. ValidateRequest is fairly vague as to what about the request is being validated.

Reply:
Updated class name and method name to indicate TargetDeviceFamily Validation.